### PR TITLE
Add unit tests & jasmine framework plus some support for vscode

### PR DESCRIPTION
**Pull request by gordonmcmullan**
Some of the helpers were starting to get sufficient logic to warrant having tests, so I added Jasmine, in line with assets-frontend, and wrote some tests for a few of them.

I also added support for vscode to run the tests, and jshint in vscode.

### CODE REVIEW

File: spec/helpers/reporters.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const SpecReporter = require('jasmine-spec-reporter').SpecReporter;
```
Comment (paul-haine) [inline spec/helpers/reporters.js]: There are semi-colons in this file, should they be removed to match other files?

File: spec/titleCaseSpec.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+describe('applying titleCase to a fragment', function() {
+var titleCase = require('../helpers/titleCase')
```
Comment (paul-haine) [inline spec/titleCaseSpec.js]: Very minor niggle, but I think this line should be indented for consistency

File: spec/helpers/reporters.js (hunk 1 above)
Comment (gordonmcmullan) [reply spec/helpers/reporters.js]: Done

File: spec/titleCaseSpec.js (hunk 2 above)
Comment (gordonmcmullan) [reply spec/titleCaseSpec.js]: Yes, you're right, indented now.

**Pull request merged**